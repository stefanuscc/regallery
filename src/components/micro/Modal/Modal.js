import styles from './Modal.module.scss'

export const Modal = (props) => {
    const {
        className="", 
        title = "", 
        content,
        isOpen = false, 
        onClose, 
        ...attrs
    } = props

    const classNames = `${styles.modal} ${!isOpen ? styles.hidden : ''} ${className}`

    return (
        <div className={classNames} {...attrs}>
            <div className={styles.dialog}>
                { title ? 
                    (<div className="header">{ title ?? "Rename Image"}</div>) 
                : ''}
                <div className="header">{ title ?? "Rename Image"}</div>
                <div className="content">{ props.children }</div>
            </div>
        </div>
    )
}

export default Modal