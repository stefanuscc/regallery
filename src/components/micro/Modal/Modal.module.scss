// Vars
@import "/src/_vars";
.modal {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 1050;
    width: 100%;
    height: 100%;
    overflow: hidden;
    outline: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: rgba($color: #000000, $alpha: .5);

    &.hidden {
        display: none;
    }

    .dialog {
        padding: 1.5rem;
        background-color: white;
    }
    
    .header {
        width: 100%;
        border-bottom: 1px solid gray;
        font-size: 18px;
        text-align: center;
        padding: 5px;
    }

    .content {
        width: 100%;
        padding: 10px 5px;
    }

    .actions {
        margin: auto;
        width: 100%;
        padding: 10px 5px;
        text-align: center;
    }

    .close {
        cursor: pointer;
        position: absolute;
        display: block;
        padding: 2px 5px;
        line-height: 20px;
        right: -10px;
        top: -10px;
        font-size: 24px;
        background: #ffffff;
        border-radius: 18px;
        border: 1px solid #cfcece;
    }
}