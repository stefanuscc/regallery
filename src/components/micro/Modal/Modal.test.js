import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Modal from './Modal';
import "../../../setupTests"

test('Modal to match snapshot', () => {
    const component = renderer.create(
        <Modal isOpen={true}>
            Hello World
        </Modal>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Modal content is persistence', () => {
    const modal = shallow(
        <Modal isOpen={true}>
            Hello World
        </Modal>
    );

    expect(modal.text()).toEqual('Hello World');
});