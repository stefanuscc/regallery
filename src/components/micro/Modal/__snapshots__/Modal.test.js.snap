// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal to match snapshot 1`] = `
<div
  className="modal  "
>
  <div
    className="dialog"
  >
    
    <div
      className="header"
    >
      
    </div>
    <div
      className="content"
    >
      Hello World
    </div>
  </div>
</div>
`;
